app/src/main/res/xml/rowkeys_qwerty0.xml

```diff
@@ -17,30 +17,58 @@
 ** limitations under the License.
 */
 -->
-
 <merge xmlns:latin="http://schemas.android.com/apk/res-auto">
-    <Row
-        latin:rowHeight="15%p"
-        >
-        <Key
-            latin:keySpec="1" />
-        <Key
-            latin:keySpec="2" />
-        <Key
-            latin:keySpec="3" />
-        <Key
-            latin:keySpec="4" />
-        <Key
-            latin:keySpec="5" />
-        <Key
-            latin:keySpec="6" />
-        <Key
-            latin:keySpec="7" />
-        <Key
-            latin:keySpec="8" />
-        <Key
-            latin:keySpec="9" />
-        <Key
-            latin:keySpec="0" />
-    </Row>
-</merge>
+    <switch>
+        <case
+            latin:showExtraChars="true"
+            latin:keyboardLayoutSetElement="alphabetManualShifted|alphabetShiftLocked|alphabetShiftLockShifted">
```

Comment:
Something is off with capitalization and it's probably here. Number row = ON, extra chars = ON.
1. Pointer position at 0 -> all letters are capital + 123 in top row
2. Type some text -> 123 still on top
3. Click shift -> !@# on top
4. Delete all text -> 123 still on top
5. Click shift -> Top row switches from 123 -> !@# -> 123 in a second

Reply:
Having numbers on top when shift was automatic is weird, but having special chars in the initial state is also weird. There won't be a correct answer here, so it's probably best to stick to what GBoard does (current). Special chars on top look weird in general, but it's also what GBoard does.

Reply:
For the 123 -> !@# -> 123 issue, what's happening is that when pressing the shift key, the state changes to alphabetManualShifted, and when shift is released it changes to alphabet. This issue of keys briefly flashing something else when pressing the shift key when it is already automatically shifted is actually an existing problem in the dvorak, azerty, lao, and thai keyboards (and probably some more). I'm thinking a new alphabetAutomaticShiftShifted state needs to be created, similar to alphabetShiftLockShifted, to allow the layout to handle it differently, but I don't really want to add that fix to this pull request.

If you're still alright with the row height change without the addition of the symbols on the number row, I can just back that part out (no longer fix #206). As a separate pull request, I can try to fix the shift state issue and then add the symbols in the number row. Alternatively, if you prefer, we could put this pull request on hold until I submit a new pull request for the fix for the shift state.

Reply:
My vote is to exclude the extra symbols in number row, or hold off till the capitalization is fixed in a separate PR.
I haven't seen this issue in AZERTY layouts before, probably because it only flashes a single key. I wonder if this issue also exists in original LatinIME.

Shift on event is on click, shift off event is on release. I'd say this is inconsistent even thou it's probably done to better support the shift double-click and long-press events, and GBoard does the same.

_alphabetManualShifted_ should probably be moved to shift release in case of uncapitalization. It isn't manually shifted yet as capitalization is done on release.

This is a separate PR/issue. If I'll have time in the weekend I'll take a look if you won't make it faster.

Reply:
I removed the symbols from the number row.